### DISCUSSION TITLE
List of Products - Price Range

### DISCUSSION THREAD

**Discussion by Lulololu** [swellstores/community#397] [Help] [answered]
Hi,

Is there a way to retrieve the Price Range from a list of Products other than fetching all Products with the pagination and, once all products obtained, calculate the price range ?
Because for long list of Products, it means that we have to wait for 10 seconds, for all our calls to the API (with a limit of 100 products for each call) to be made - when we would need the Price Range at the loading of the page.

Example of the function we have right now:
```

async function fetchAllCategoryProducts(
  categorySlug: string,
): Promise<Array<ImprovedProduct>> {
  let allProducts: Array<ImprovedProduct> = [];
  let hasNextPage = true;
  let currentPage = 1;

  while (hasNextPage) {
    const { count, results } = await swellClient.products.list({
      category: categorySlug,
      limit: PRODUCTS_CATEGORY_FETCH_NUMBER_MAX,
      page: currentPage,
    });

    allProducts = [...allProducts, ...results];
    hasNextPage = allProducts.length < count;
    currentPage++;
  }

  return allProducts;
}
```

Thanks for your help,

**Comment by sags95** [comment, marked as answer]
Unfortunately with the Frontend API there is no method to retrieve a price range, although I agree that this would be useful. It's something I can definitely pass on to the team.

**Reply by sags95** [reply]
@Lulololu One thing that might solve for this is to create a filter list with `await swell.products.filters(products)`, where products is a list of products. It should return a `price` filter that includes a range like:

```
  {
    "id": "price",
    "label": "Price",
    "type": "range",
    "interval": 10,
    "options": [
      {
        "value": 10,
        "label": "$10.00"
      },
      {
        "value": 100,
        "label": "$100.00"
      }
    ]
  }
```

**Comment by Lulololu** [comment]
Thank you @sags95
